controllers/csiscaleoperator_controller.go

```diff
@@ -665,24 +812,47 @@ func (r *CSIScaleOperatorReconciler) reconcileClusterRole(instance *csiscaleoper
 			Namespace: cr.Namespace,
 		}, found)
 		if err != nil && errors.IsNotFound(err) {
-			logger.Info("creating a new ClusterRole", "Name", cr.GetName())
+			logger.Info("Creating a new ClusterRole", "Name", cr.GetName())
 			err = r.Client.Create(context.TODO(), cr)
 			if err != nil {
+				message := "Failed to create ClusterRole " + cr.GetName()
+				logger.Error(err, message)
+				// TODO: Add event.
+				meta.SetStatusCondition(&crStatus.Conditions, metav1.Condition{
+					Type:    string(config.StatusConditionSuccess),
+					Status:  metav1.ConditionFalse,
+					Reason:  string(csiv1.ResourceCreateError),
+					Message: message,
+				})
 				return err
 			}
 		} else if err != nil {
-			logger.Error(err, "failed to get ClusterRole", "Name", cr.GetName())
+			message := "Failed to get ClusterRole " + cr.GetName()
+			logger.Error(err, message)
+			// TODO: Add event.
+			meta.SetStatusCondition(&crStatus.Conditions, metav1.Condition{
+				Type:    string(config.StatusConditionSuccess),
+				Status:  metav1.ConditionFalse,
+				Reason:  string(csiv1.ResourceReadError),
+				Message: message,
+			})
 			return err
 		} else {
 			err = r.Client.Update(context.TODO(), cr)
 			if err != nil {
-				logger.Error(err, "failed to update ClusterRole", "Name", cr.GetName())
+				message := "Failed to update ClusterRole " + cr.GetName()
+				logger.Error(err, message)
+				// TODO: Add event.
+				meta.SetStatusCondition(&crStatus.Conditions, metav1.Condition{
+					Type:    string(config.StatusConditionSuccess),
+					Status:  metav1.ConditionFalse,
+					Reason:  string(csiv1.ResourceUpdateError),
+					Message: message,
+				})
 				return err
 			}
 		}
 	}
-
-	logger.Info("exiting reconcileClusterRole()")
```

Comment:
Don't remove this log. It can help in debugging.

Reply:
Added back.